doozerlib/olm/bundle.py

```diff
@@ -0,0 +1,450 @@
+            rc, out, err = exectools.cmd_gather(
+                'rhpkg{}container-build --nowait --target {}'.format(self.rhpkg_opts, self.target)
+            )
+
+        if rc != 0:
+            msg = 'Unable to create brew task: rc={} out={} err={}'.format(rc, out, err)
+            self.runtime.logger.info(msg)
+            return False
+
+        self.task_url = re.search(r'Task info:\s(.+)', out).group(1)
+        self.task_id = re.search(r'Created task:\s(\d+)', out).group(1)
+        return True
+
+    def watch_bundle_container_build(self):
+        """Log brew task URL and eventual task states until task completion (or failure)
+
+        :return bool True if brew task was successfully completed, False otherwise
+        """
+        self.runtime.logger.info('Build running: {}'.format(self.task_url))
+        error = brew.watch_task(
+            self.runtime.group_config.urls.brewhub,
+            self.runtime.logger.info,
+            self.task_id,
+            threading.Event()
+        )
+        if error:
+            self.runtime.logger.info(error)
+            return False
+        return True
+
+    def find_and_replace_image_references_by_sha(self, contents):
+        """Search image references (<registry>/<image>:<tag>) on given contents (usually YAML),
+        replace them with corresponding (<registry>/<image>@<sha>) and collect such replacements to
+        list them as "relatedImages" under "spec" section of contents (should it exist)
+
+        :param string contents: File contents that potentially contains image references
+        :return string: Same contents, with aforementioned modifications applied
+        """
+        found_images = {}
+
+        def collect_replaced_image(match):
+            image = '{}/{}@{}'.format(
+                'registry.redhat.io',  # hardcoded until appregistry is dead
+                match.group(1).replace('openshift/', 'openshift4/'),
+                self.fetch_image_sha('{}:{}'.format(match.group(1), match.group(2)))
+            )
+            key = u'{}'.format(re.search(r'([^\/]+)\/(.+)', match.group(1)).group(2))
+            found_images[key] = u'{}'.format(image)
+            return image
+
+        new_contents = re.sub(
+            r'{}\/([^:]+):([^\'"\s]+)'.format(self.operator_csv_config['registry']),
+            collect_replaced_image,
+            contents,
+            flags=re.MULTILINE
+        )
+
+        return self.append_related_images_spec(new_contents, found_images)
+
+    def fetch_image_sha(self, image):
+        """Get corresponding SHA of given image (using skopeo)
+
+        OCP 4.3+ supports "manifest-lists", which is a SHA that doesn't represent an actual image,
+        but a list of images per architecture instead. OCP 4.3+ is smart enough to read that list
+        and pick the correct architecture.
+
+        Unfortunately, OCP 4.2 is multi-arch and does not support manifest-lists. It is still unclear
```

Comment:
Actually it does support manifest lists now, but it didn't always.

Reply:
@sosiouxme does that mean we can just use manifest-list SHAs now??
if so, that's **GREAT** news! that means we can change [this line](https://github.com/openshift/ocp-build-data/blob/openshift-4.2/group.yml#L12) and be done with it.

Reply:
that's the outcome I was trying to get to. but i'm not sure we got there.
possibly we should just do appregistry with 4.2 until it's EOL? i'm not sure that's an option though.

Reply:
4.2 will be EOL by the time we're shipping these; guess we can ignore it unless we need to rebuild 4.2 operators post-EOL for some reason :vomiting_face: